### PR TITLE
#1089 checkstyle

### PR DESCRIPTION
**Pull request by TheRealMarnes**
#1089

### CODE REVIEW

**Comment by TheRealMarnes** [comment]
Think I might use this instead

https://code.revelc.net/formatter-maven-plugin/

https://stackoverflow.com/questions/24462956/maven-code-formatter-plugin-ignoring-my-java-files

Doing it manually is a PITA and not very futureproof

**Comment by TheRealMarnes** [comment]
this is a good start for now

**Comment by TheRealMarnes** [comment]
Hmm, strange, it's ignoring tests...

**Comment by TheRealMarnes** [comment]
Let's ban the use of `@Test.expected` in favor of `assretThatThrownBy` since it's less likely to lead to accidents and gives more consistent use of assertj?

File: core/src/main/java/org/jdbi/v3/core/OnDemandExtensions.java [hunk 1]
```diff
@@ -30,8 +30,7 @@
             EQUALS_METHOD = Object.class.getMethod("equals", Object.class);
             HASHCODE_METHOD = Object.class.getMethod("hashCode");
             TOSTRING_METHOD = Object.class.getMethod("toString");
-        }
-        catch (NoSuchMethodException wat) {
+        } catch (NoSuchMethodException wat) {
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/core/OnDemandExtensions.java]: this is my favorite argument name

File: core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java [hunk 2]
```diff
@@ -132,9 +132,12 @@ public static Type resolveMapEntryType(Type keyType, Type valueType) {
...
     private static <K, V> Type resolveMapEntryType(TypeToken<K> keyType, TypeToken<V> valueType) {
-        return new TypeToken<Map.Entry<K, V>>() {}
-                .where(new TypeParameter<K>() {}, keyType)
-                .where(new TypeParameter<V>() {}, valueType)
+        return new TypeToken<Map.Entry<K, V>>() {
+        }
+                .where(new TypeParameter<K>() {
+                }, keyType)
+                .where(new TypeParameter<V>() {
+                }, valueType)
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java]: ugh

Comment (TheRealMarnes) [reply core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java]: yeah, unfortunately I couldn't find a way in checkstyle to ignore {}... at the same time, {} is actually discouraged anyway as an empty block. It's not pretty but it makes sense ¯\\\_(ツ)\_/¯

File: core/src/main/java/org/jdbi/v3/core/statement/BaseStatement.java [hunk 3]
```diff
@@ -24,26 +24,22 @@
...
-    BaseStatement(Handle handle)
-    {
+    BaseStatement(Handle handle) {
         this.handle = handle;
         this.ctx = new StatementContext(
                 handle.getConfig().createCopy(), handle.getExtensionMethod());
-    }
-
-    {
+    } {
```
Comment (qualidafial) [inline core/src/main/java/org/jdbi/v3/core/statement/BaseStatement.java]: Why don't we just fold this initializer into the constructor? i.e. remove line 39

Comment (TheRealMarnes) [reply core/src/main/java/org/jdbi/v3/core/statement/BaseStatement.java]: done

File: core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java (hunk 2 above)
Comment (TheRealMarnes) [reply core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java]: working on this

File: sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/ResultReturner.java [hunk 4]
```diff
@@ -155,197 +158,161 @@ protected Type elementType(StatementContext ctx) {
...
-
-        ResultIterableReturner(Type returnType)
-        {
-            // extract T from Query<T>
-            elementType = GenericTypes.findGenericParameter(returnType, ResultIterable.class)
-                    .orElseThrow(() -> new IllegalStateException(
-                            "Cannot reflect ResultIterable<T> element type T in method return type " + returnType));
-        }
-
+    static class VoidReturner extends ResultReturner {
```
Comment (qualidafial) [inline sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/ResultReturner.java]: Why reorder the inner classes? Makes it difficult to review

Comment (TheRealMarnes) [reply sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/ResultReturner.java]: I've widely seen it considered good practice to define inner classes/interfaces/enums at the bottom of the parent class to keep them out of the way of the class logic

**Comment by TheRealMarnes** [comment]
jesus christ, people have committed a shitload of code smells in the test code especially... like declaring checked exceptions that are never thrown for no reason

File: sqlobject/src/test/java/org/jdbi/v3/sqlobject/statement/TestSqlScripts.java [hunk 5]
```diff
@@ -36,10 +36,10 @@ public void testCreateTable() {
 
     private interface Scriptacular {
         // tag::scripts[]
-        @SqlScript("CREATE TABLE <name> ( pk int primary key )")
+        @SqlScript("CREATE TABLE <name> (pk int primary key)")
         void createTable(@Define String name);
 
-        @SqlScript("INSERT INTO cool_table VALUES ( 5 ), ( 6 ), ( 7 )")
+        @SqlScript("INSERT INTO cool_table VALUES (5), (6), (7)")
```
Comment (qualidafial) [inline sqlobject/src/test/java/org/jdbi/v3/sqlobject/statement/TestSqlScripts.java]: I'm surprised/amazed checkstyle is finding these spaces _inside_ the strings.

File: src/build/checkstyle.xml [hunk 6]
```diff
@@ -66,6 +66,27 @@
...
+		<module name="NoFinalizer"/>
+		<module name="OneStatementPerLine"/>
+		<module name="OneTopLevelClass"/>
+		<module name="OuterTypeFilename"/>
+		<module name="ParameterAssignment"/>
+		<module name="StringLiteralEquality"/>
+		<module name="SuperClone"/>
+		<module name="SuperFinalize"/>
+		<!--<module name="TrailingComment"/>-->
+		<module name="VariableDeclarationUsageDistance"/>
```
Comment (qualidafial) [inline src/build/checkstyle.xml]: What's the basis for this ruleset? Is this just Sun/Oracle Java conventions?

File: sqlobject/src/test/java/org/jdbi/v3/sqlobject/statement/TestSqlScripts.java (hunk 5 above)
Comment (TheRealMarnes) [reply sqlobject/src/test/java/org/jdbi/v3/sqlobject/statement/TestSqlScripts.java]: Hmm, that's my bad, sorry

File: src/build/checkstyle.xml (hunk 6 above)
Comment (TheRealMarnes) [reply src/build/checkstyle.xml]: Checkstyle offerings and what I experience as the concensus among java devs at the clients I've worked for... Do you mean some specific rules or the whole lot aside from the few mentioned explicitly so far?

File: sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/ResultReturner.java (hunk 4 above)
Comment (qualidafial) [reply sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/ResultReturner.java]: I get that, but why change the order of the inner classes themselves?

Comment (TheRealMarnes) [reply sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/ResultReturner.java]: I may not really have paid attention to that, having to sift through hundreds of violations ;_;

File: sqlobject/src/test/java/org/jdbi/v3/sqlobject/statement/TestSqlScripts.java (hunk 5 above)
Comment (qualidafial) [reply sqlobject/src/test/java/org/jdbi/v3/sqlobject/statement/TestSqlScripts.java]: I'm ok with normalizing SQL formatting so long as it's not something the test cares about.

File: src/build/checkstyle.xml [hunk 7]
```diff
@@ -46,7 +44,12 @@
 		<module name="ParenPad"/>
 		<module name="TypecastParenPad"/>
 		<module name="WhitespaceAfter"/>
-		<module name="WhitespaceAround"/>
+		<module name="WhitespaceAround">
+            <property name="allowEmptyMethods" value="true"/>
+            <property name="allowEmptyConstructors" value="true"/>
+            <property name="allowEmptyLambdas" value="true"/>
+            <property name="allowEmptyTypes" value="true"/>
+        </module>
```
Comment (qualidafial) [inline src/build/checkstyle.xml]: Mixed tabs/spaces

File: src/build/checkstyle.xml (hunk 6 above)
Comment (qualidafial) [reply src/build/checkstyle.xml]: I was looking at the whole.

Ok so this is drawn from experience / personal taste. That's fine, I think I agree with the changes you've made.

**Comment by TheRealMarnes** [comment]
Apparently checkstyle doesnt run against tests by default... I'm thinking of excluding tests from some of the rules because this is becoming a very huge, risky merge :/

**Comment by qualidafial** [comment]
You wanna merge this one and finish the rest in a separate PR?

**Comment by TheRealMarnes** [comment]
If you're ok with the content *so far*, yeah, sure. Would make a solid base to continue from. I'm worried about merges with currently existing branches.

**Comment by qualidafial** [comment]
Thanks @TheRealMarnes!

File: core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java (hunk 2 above)
Comment (stevenschlansker) [reply core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java]: I don't like this change, stylistically.  I think it makes the code much less readable.  To me, the `new TP<K>() {}` construction, bizarre as it is, reads as a single "thing" and shouldn't get line split.

Comment (TheRealMarnes) [reply core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java]: I don't like it either, I fixed this in part 2: https://github.com/jdbi/jdbi/pull/1100

Comment (TheRealMarnes) [reply core/src/main/java/org/jdbi/v3/core/generic/GenericTypes.java]: Specifically https://github.com/jdbi/jdbi/pull/1100/commits/685f4159a518e8475a84ec5444d35c0b9d5ddfb0

**Pull request merged**